enterprise/backend/src/metabase_enterprise/serialization/api.clj

```diff
@@ -0,0 +1,237 @@
+          (.removeLogger config (.getName logger))
+          (.stop appender)
+          ;; this method is only present in AbstractConfiguration
+          (.removeAppender config (.getName appender))
+          (.updateLoggers (logger/context)))))))
+
+;;; Logic
+
+(defn- serialize&pack ^File [opts]
+  ;; TODO: how to get time in local timezone?
+  (let [id       (format "%s-%s"
+                         (u/slugify (public-settings/site-name))
+                         (u.date/format "YYYY-MM-dd_HH-mm" (t/zoned-date-time)))
+        path     (io/file parent-dir id)
+        dst      (io/file (str (.getPath path) ".tar.gz"))
+        log-file (io/file path "export.log")]
+    (with-open [_logger (make-logger 'metabase-enterprise.serialization log-file)]
+      (try                              ; try/catch inside logging to log errors
+        (-> (extract/extract opts)
+            (storage/store! path))
+        ;; not removing storage immediately to save some time before response
+        (compress-tgz path dst)
+        (catch Exception e
+          (log/error e "Error during serialization"))))
+    {:archive  (when (.exists dst)
+                 dst)
+     :log-file (when (.exists log-file)
+                 log-file)
+     :callback (fn []
+                 (when (.exists path)
+                   (run! io/delete-file (reverse (file-seq path))))
+                 (when (.exists dst)
+                   (io/delete-file dst)))}))
+
+(defn- unpack&import [^File file & [size]]
+  (let [dst      (io/file parent-dir (apply str (repeatedly 20 #(char (+ 65 (rand-int 26))))))
+        log-file (io/file dst "import.log")]
+    (with-open [_logger (make-logger 'metabase-enterprise.serialization log-file)]
+      (try                              ; try/catch inside logging to log errors
+        (log/infof "Serdes import, size %s" size)
+        (let [path (uncompress-tgz file dst)]
+          (serdes/with-cache
+            (-> (v2.ingest/ingest-yaml (.getPath (io/file dst path)))
+                (v2.load/load-metabase! {:abort-on-error true}))))
+        (catch Exception e
+          (log/error e "Error during serialization"))))
+    {:log-file log-file
+     :callback #(when (.exists dst)
+                  (run! io/delete-file (reverse (file-seq dst))))}))
+
+(comment
+  ;; dump on disk and see that it works
+  (serialize&pack {:no-data-model true :no-settings true :targets [["Collection" 31409]]})
+  (unpack&import (io/file "metabase_test.tar.gz")))
+
+;;; HTTP API
+
+(api/defendpoint POST "/export"
+  "Serialize and retrieve Metabase data."
+  [:as {{:strs [all-collections collection settings data-model field-values database-secrets]
+         :or   {settings        true
+                data-model      true
+                all-collections true}}
+        :query-params}]
+  {collection       [:maybe [:or ms/PositiveInt [:sequential ms/PositiveInt]]]
+   all-collections  [:maybe ms/BooleanValue]
```

Comment:
```suggestion
   all_collections  [:maybe ms/BooleanValue]
```

we use underscore pattern across APIs, I think we should be consistent here.

Reply:
also, why don't we use the same parameter names that we use in the cmd?
 `include-database-secrets` is a better name than `database-secrets` IMO.

Reply:
Oh, do we? That's kind of against the usual patterns of URLs... I thought we're doing underscores in JSON since it's easier (no quotes around) to use as keys in JS, but for query parameters to have underscore... it's like having underscores in command line options.

As for `include-database-secrets` — I think it's named like that because it's an argument-less option, you don't say `--include-database-secrets=true`, as you do in URL.

Reply:
ah to be clear, I think we should keep the same name, but change from dash to underscore.

so `include-database-secrets` will be `include_database_secrets`